web/settings/disk_usage.go

```diff
@@ -37,5 +38,11 @@ func diskUsage(c echo.Context) error {
 		return err
 	}
 
-	return jsonapi.Data(c, http.StatusOK, &apiDiskUsage{used}, nil)
+	var result = &apiDiskUsage{used}
+
+	if err = permissions.Allow(c, permissions.GET, result); err != nil {
+		return err
+	}
```

Comment:
I'd prefer to check the permission before calling `vfs.DiskUsage`. Something like that:

```go
func diskUsage(c echo.Context) error {
    var result = &apiDiskUsage{}
    if err = permissions.Allow(c, permissions.GET, result); err != nil {
        return err
    }

    instance := middlewares.GetInstance(c)
    used, err := vfs.DiskUsage(instance)
    if err != nil {
        return err
    }

    result.Used = used
    return jsonapi.Data(c, http.StatusOK, result, nil)
}
```